routes/deductions/deductions.controller.js

```diff
@@ -0,0 +1,50 @@
+const { validationResult } = require('express-validator')
+const { errorArray2ErrorObject } = require('./../../utils.js')
+//const { loginSchema, sinSchema } = require('./../../formSchemas.js')
+//const API = require('../../api')
+
+module.exports = function (app) {
+  // redirect from "/deductions" → "/login/code", you really shouldn't be here (Unless we add a "info" page)
+  app.get('/deductions', (req, res) => res.redirect('/login/code'))
+
+  app.get('/deductions/rrsp', (req, res) => res.render('deductions/rrsp', { data: req.session || {} }))
+  app.post('/deductions/rrsp', postRRSP)
+
+}
+
+//POST functions that handle setting the login data in the session and handle redirecting to the next page or sending an error to the client.
+//Note that this is not the only error validation, see routes defined above.
+const validateRedirect = req => {
+  let redirect = req.body.redirect || null
+  if (!redirect) {
+    throw new Error(`[POST ${req.path}] 'redirect' parameter missing`)
+  }
+  return redirect
+}
+
+const postRRSP = (req, res) => {
+  const redirect = validateRedirect(req, res)
+
+  const errors = validationResult(req)
+
+  //If rrsp is not set, set it to null
+  let rrsp = req.body.rrsp || null
+  req.session.deductions.rrspClaim = rrsp
```

Comment:
We shouldn't be updating the session until we're past the `(!errors.isEmpty())` check. But otherwise, yeah, this is the way to update user session stuff.

Reply:
One thing to note is that I've stubbed out all the user session deductions/claims.

So, two things.

1. [Update `"claims":` to `"deductions":`](https://github.com/cds-snc/cra-claim-tax-benefits/blob/master/api/user.json#L51)

2. Since lots of them are opt-in, I've added the `{whatever}Claim` key meant to be true/false (or null as the default), but then we can add more keys as we need for the data the user is actually submitting. In this case, [if `rrspClaim` is true](https://github.com/cds-snc/cra-claim-tax-benefits/blob/master/api/user.json#L53), then we should expect [an `rrspAmount`](https://github.com/cds-snc/cra-claim-tax-benefits/blob/master/api/user.json#L54) as well.

Reply:
done